platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+#include "coding/file_reader.hpp"
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+
+/// @return directory where binary resides, including slash at the end
+static bool GetBinaryFolder(string & outPath)
+{
+  int mib[4];
+  mib[0] = CTL_KERN;
+  mib[1] = KERN_PROC;
+  mib[2] = KERN_PROC_PATHNAME;
+  mib[3] = -1;
+  char path[4096];
+  size_t cb = ARRAY_SIZE(path);
+  if (::sysctl(mib, 4, path, &cb, NULL, 0) == 0) {
+    outPath = path;
+    outPath.erase(outPath.find_last_of('/') + 1);
+    return true;
+  }
+  return false;
+}
+
+Platform::Platform()
+{
+  // init directories
+  string path;
+  CHECK(GetBinaryFolder(path), ("Can't retrieve path to executable"));
+
+  char const * homePath = ::getenv("HOME");
+  string const home(homePath ? homePath : "");
+
+  m_settingsDir = home + "/.config/MapsWithMe";
+
+  if (!IsFileExistsByFullPath(m_settingsDir + SETTINGS_FILE_NAME))
+  {
+    mkdir((home + "/.config/").c_str(), 0755);
+    mkdir(m_settingsDir.c_str(), 0755);
+  }
+
+  m_writableDir = home + "/.local/share/MapsWithMe";
+  mkdir((home + "/.local/").c_str(), 0755);
+  mkdir((home + "/.local/share/").c_str(), 0755);
+  mkdir(m_writableDir.c_str(), 0755);
+
+  char const * resDir = ::getenv("MWM_RESOURCES_DIR");
+  if (resDir)
```

Comment:
This function looks like platform/platform_linux.cpp. It's not good to just duplicate it's code here. Please, reuse old code instead of copy-and-paste.